### PR TITLE
Implement functionality to save and display most visited websites

### PR DESCRIPTION
**Pull request by HK-Gupta**
Created functionality to add and display the most visited websites on the browser home page.

### CODE REVIEW

File: lib/app_bar/webview_tab_app_bar.dart [hunk 1]
```diff
@@ -77,6 +80,85 @@ class _WebViewTabAppBarState extends State<WebViewTabAppBar>
     });
   }
 
+
+  Future<void> _saveMostVisitedWebsite(String urlString) async {
```
Comment (32bitThug) [inline lib/app_bar/webview_tab_app_bar.dart]: move these  functions to end of the file or create another file for them and use these functions here

File: lib/app_bar/webview_tab_app_bar.dart [hunk 2]
```diff
@@ -394,6 +501,7 @@ class _WebViewTabAppBarState extends State<WebViewTabAppBar>
                             onPressed: () {
                               setState(() {
                                 if (favorite != null) {
+                                  // _saveMostVisitedWebsite("www.leetcode.com");
```
Comment (32bitThug) [inline lib/app_bar/webview_tab_app_bar.dart]: remove this line if not using it

File: lib/app_bar/webview_tab_app_bar.dart [hunk 3]
```diff
@@ -145,18 +228,18 @@ class _WebViewTabAppBarState extends State<WebViewTabAppBar>
...
               }
 
               if (webViewController != null) {
-                webViewController.loadUrl(urlRequest: URLRequest(url: url));
+                // Load the URL in the existing WebView
+                await webViewController.loadUrl(urlRequest: URLRequest(url: url));
               } else {
+                // Add a new tab and load the URL
                 addNewTab(url: url);
-                webViewModel.url = url;
```
Comment (32bitThug) [inline lib/app_bar/webview_tab_app_bar.dart:159]: you should update the webviewModel, do not remove this line

**Review by 32bitThug — CHANGES_REQUESTED** [review]
make these changes

**Comment by HK-Gupta** [comment]
I made the changes you suggested, but I'm still encountering a SonarQube error.

**Comment by 32bitThug** [comment]
> I made the changes you suggested, but I'm still encountering a SonarQube error.

Are you able to build the app in your local machine without any build errors?

**Pull request merged**